tests/test_pygtracker.py

```diff
@@ -26,6 +28,71 @@ def test_version():
 
 # Start tests for rank_students
 
+# tests that the user inputs the right number of students
+def test_rank_students_num_of_students_match():
+    tracker = generate_input_calculate_final_grade()
+    assert (tracker.rank_students(n=4).shape[0]) == 4, "Number of students don't match"
+
+
+# raises an error when a course id is not a part of the dataset
+def test_rank_students_course_input_not_exist():
+    tracker = generate_input_calculate_final_grade()
+    with raises(NameError):
+        tracker.rank_students(course_id="500", n=4)
+
+
+def test_rank_students_ascending_input_not_bool():
+    gradetracker = pygtracker.GradeTracker()
+    with raises(TypeError):
+        gradetracker.rank_students(course_id="511", ascending="True")
+
+
+def test_rank_students_n_input_not_integer():
+    gradetracker = pygtracker.GradeTracker()
+    with raises(TypeError):
+        gradetracker.rank_students(n="4")
+
+
+def test_rank_students_course_input_not_integer():
+    gradetracker = pygtracker.GradeTracker()
+    with raises(TypeError):
+        gradetracker.rank_students(course_id=511)
+
+
+# dataframe is equal to an example dataframe - when if statement is False
+def test_rank_students_one_course_df_equal():
+    tracker = generate_input_calculate_final_grade()
+    rank_df = tracker.rank_students(course_id="511", n=1, ascending=False)
+
+    example_df = pd.DataFrame(
+        {"student_id": "joel", "grade": 90.82, "rank": 1.0}, index=[3]
+    )
+
+    assert_frame_equal(rank_df, example_df)
+
+
+# dataframe is equal to an example dataframe - when if statement is True
+def test_rank_students_all_courses_df_equal():
+    tracker = generate_input_calculate_final_grade()
+    rank_df = tracker.rank_students(course_id="all", n=1, ascending=False)
+
+    example_df = pd.DataFrame(
+        {"student_id": "joel", "grade": 91.81, "rank": 1.0}, index=[0]
+    )
+
+    assert_frame_equal(rank_df, example_df)
+
+
+# check the columns of the dataframe that come of the function
+def test_rank_students_columns_names_match():
```

Comment:
Technically this test is redundant because when you run the test right above, it should check for the column names as well if i'm not mistaken

Reply:
yes - agreed. Will remove and add another test.